mteb/evaluation/evaluators/RerankingEvaluator.py

```diff
@@ -179,39 +189,82 @@ def compute_metrics_individual(self, model):
             )
             docs_emb = np.asarray(encode_corpus_func(docs, batch_size=self.batch_size))
 
-            scores = self._compute_metrics_instance(query_emb, docs_emb, is_relevant)
+            sim_scores = self._compute_sim_scores_instance(query_emb, docs_emb)
+            scores = self._compute_metrics_instance(sim_scores, is_relevant)
+            conf_scores = confidence_scores(sim_scores)
+
             all_mrr_scores.append(scores["mrr"])
             all_ap_scores.append(scores["ap"])
+            all_conf_scores.append(conf_scores)
 
         mean_ap = np.mean(all_ap_scores)
         mean_mrr = np.mean(all_mrr_scores)
 
-        return {"map": mean_ap, "mrr": mean_mrr}
+        # Compute nAUCs
+        naucs_map = self._compute_nAUCs(all_conf_scores, all_ap_scores, "map")
+        naucs_mrr = self._compute_nAUCs(all_conf_scores, all_mrr_scores, "mrr")
 
-    def _compute_metrics_instance(self, query_emb, docs_emb, is_relevant):
-        """Computes metrics for a single instance = (query, positives, negatives)
+        return {**{"map": mean_ap, "mrr": mean_mrr}, **naucs_map, **naucs_mrr}
+
+    def _compute_sim_scores_instance(self, query_emb, docs_emb):
```

Comment:
I would move the types to the function def. -- I know it was like that beforehand but we are slowly moving it.

Reply:
Done!